### ISSUE TITLE
Writing custom data to FIT files

### ISSUE THREAD

**Issue by FunkEater** [hammerheadnav/karoo-ext#13]
Enhance functionality to allow writing custom data to FIT files, enabling features like the potential integration of “MyBikeTraffic.com,” for instance.

https://support.hammerhead.io/hc/en-us/community/posts/9398520826907-Feature-Request-MyBikeTraffic-Integration

**Comment by brentnd** [comment]
Including custom FIT file data is something that we've discussed and planned for inclusion but there's some further thought needed to flesh out a flexible and powerful solution.

Can you provide more details about what part of the FIT file you'd want to augment. For example, adding fields to our existing record messages, writing your own events, defining developer fields/messages, etc.

**Comment by lockevod** [comment]
Hello! I think this can be very useful. For example, I’m developing a VO2max custom data type in kPower (this is a virtual power meter for Karoo).  It’s ok but if u can have this info also in the fit file

**Comment by brentnd** [comment]
@lockevod  Since it seems like VO2 max isn't part of the standard FIT library, do you intend to use a custom field? Does this apply to summary records only or all RecordMesgs?

**Comment by lockevod** [comment]
Yes a custom field and I think VO2 is useful for summary records.
